examples/webgpu_tsl_editor.html

```diff
@@ -126,15 +126,21 @@
 					const editorDOM = document.getElementById( 'source' );
 					const resultDOM = document.getElementById( 'result' );
 
-					const tslCode = `// Simple example
+					const tslCode = `// Simple uv.x animation
 
-const { texture, uniform, vec4 } = TSL;
+const { texture, uniform, vec2, vec4, uv, oscSine, timerLocal } = TSL;
 
 //const samplerTexture = new THREE.Texture();
 const samplerTexture = new THREE.TextureLoader().load( './textures/uv_grid_opengl.jpg' );
+samplerTexture.wrapS = THREE.RepeatWrapping;
+//samplerTexture.wrapT = THREE.RepeatWrapping;
+
+const timer = timerLocal( .5 ); // .5 is speed
+const uv0 = uv();
+const animateUv = vec2( uv0.x.add( oscSine( timer ) ), uv0.y );
```

Comment:
By the way... Could we try adding support for something like `vectorVariable.x.addAssign( something )`, based on how we have that for VarNodes?

Reply:
I like `node.x.assign/addAssign` more because it better mirrors the shader code -- i.e. `var.x = something` or `var.x += something`. It also feels more understandable for me.
(And to add all the `*Assign` things we should only add `assign` support for SplitNode -- other `*Assigns` will be supported automatically)

Reply:
It looks good 👍

Reply:
Merging now, anyway better to create a new PR for it.

Reply:
Would you rather work on it, or I work on it?

Reply:
Hmm... worries a bit the users use just `uv0.x.addAssign( oscSine( timer ) );` instead of `const animateUv = uv0.x.addAssign( oscSine( timer ) );`
```js

const animateUv = uv0.x.addAssign( oscSine( timer ) );
```

Reply:
Maybe we should just note somewhere that they should either do `const animateUv = temp( uv() ); animateUv.x.addAssign( oscSine( timer ) )` or `const uv0 = uv(); const animateUv = uv0.x.addAssign( oscSine( timer ) )`.

> Would you rather work on it, or I work on it?

I think you can implement this better than me 👍

Reply:
```js
// Example 1
const animateUv = temp( uv() ); 
animateUv.x.addAssign( oscSine( timer ) );

// Example 2
const uv0 = uv(); 
const animateUv = uv0.x.addAssign( oscSine( timer ) );

// Example 3
const uv0 = uv(); 
const animateUv = uv0.addX( oscSine( timer ) );
```

It seems that we have 3 options at now, `.set*()`, `.add*()` still seems more simple for me.

> I think you can implement this better than me 👍

Thanks, but you'd do a great job all the same.

Reply:
> It seems that we have 3 options at now, .set*(), .add*() still seems more simple for me.

I think we can try to support both first two options -- we can check in SplitNode whether the node it's acting on is VarNode, and if it is not it automatically makes it a such.